### PR TITLE
Automatic actions

### PR DESCRIPTION
**Pull request by tcezard**
Transfer of the automatic review from EGCGProjectManagement to Reporting app actions

### CODE REVIEW

File: etc/schema.yaml [hunk 1]
```diff
@@ -55,8 +55,8 @@ run_elements:
     review_comments: {          type: 'string',  required: False }
     review_date: {              type: 'datetime', required: False }
     useable: {                  type: 'string',  required: False, allowed: ['not marked', 'yes', 'no' ], default: 'not marked' }
-    useable_comments: {         type: 'multilinestring',  required: False }
```
Comment (mwhamgenomics) [inline etc/schema.yaml]: We should change this in samples as well (line 138)

File: rest_api/actions/__init__.py [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
 
-from rest_api.actions.reviews import start_run_review
-
+from rest_api.actions.reviews import start_run_review, automatic_run_review, automatic_sample_review
 
 function_mapping = {
-    'run_review': start_run_review
+    'run_review': start_run_review,
+    'automatic_run_review': automatic_run_review,
+    'automatic_sample_review': automatic_sample_review
```
Comment (mwhamgenomics) [inline rest_api/actions/__init__.py]: This will conflict with #121, so needs merged carefully

File: rest_api/actions/automatic_review.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    def current_time(self):
+        return datetime.datetime.now().strftime(settings.DATE_FORMAT)
+
+    @property
+    def cfg(self):
+        raise NotImplementedError
+
+    @cached_property
+    def _summary(self):
+        raise NotImplementedError
```
Comment (mwhamgenomics) [inline rest_api/actions/automatic_review.py]: This is redefined below. Which is the intended one?

File: rest_api/actions/automatic_review.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    def __init__(self, run_id):
+        self.run_id = run_id
+        lanes = _aggregate('run_elements', queries.run_elements_group_by_lane, request_args={'match':'{"run_id": "%s"}' % run_id})
+        if not lanes:
+            raise ValueError('No data found for run id %s.' % run_id)
+        self.lane_reviewers = [LaneReviewer(lane) for lane in lanes]
+
+    @cached_property
+    def _summary(self):
+        return [r.report() for r in self.lane_reviewers]
```
Comment (mwhamgenomics) [inline rest_api/actions/automatic_review.py]: Unresolved reference - `LaneReviewer.report` has been lost somewhere in the transfer from Project-Management.

**Comment by tcezard** [comment]
The main thing that has changed is how the data is extracted from the database.
Also the Reviewer will only receive one run/sample at a time.

File: rest_api/actions/automatic_review.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    def __init__(self, run_id):
+        self.run_id = run_id
+        lanes = _aggregate('run_elements', queries.run_elements_group_by_lane, request_args={'match':'{"run_id": "%s"}' % run_id})
+        if not lanes:
+            raise ValueError('No data found for run id %s.' % run_id)
+        self.lane_reviewers = [LaneReviewer(lane) for lane in lanes]
+
+    @cached_property
+    def _summary(self):
+        return [r._summary for r in self.lane_reviewers]
```
Comment (mwhamgenomics) [inline rest_api/actions/automatic_review.py]: `RunReviewer._summary` doesn't seem to be called anywhere, so I think this can be removed.

**Pull request closed without merging**